crates/ra_hir_ty/src/db.rs

```diff
@@ -77,6 +77,8 @@ pub trait HirDatabase: DefDatabase {
     #[salsa::interned]
     fn intern_type_ctor(&self, type_ctor: TypeCtor) -> crate::TypeCtorId;
     #[salsa::interned]
+    fn intern_type_param_id(&self, param_id: TypeParamId) -> GlobalTypeParamId;
```

Comment:
Why exactly do we need this? `TypeParamId` is also just an ID (although a bigger one), so interning doesn't make a big difference I think?

Reply:
TypeParamId are two IDs which we have to fit into one number for Chalk :confused: